internal/appsec/listener/graphqlsec/graphql.go

```diff
@@ -73,8 +72,11 @@ func newWafEventListener(wafHandle *waf.Handle, cfg *config.Config, limiter limi
 // NewWAFEventListener returns the WAF event listener to register in order
 // to enable it.
 func (l *wafEventListener) onEvent(request *types.RequestOperation, _ types.RequestOperationArgs) {
-	wafCtx := waf.NewContextWithBudget(l.wafHandle, l.config.WAFTimeout)
-	if wafCtx == nil {
+	wafCtx, err := l.wafHandle.NewContextWithBudget(l.config.WAFTimeout)
+	if err != nil {
+		log.Debug("appsec: could not create budgeted WAF context: %v", err)
+	}
+	if wafCtx == nil || err != nil {
```

Comment:
This one looks hella confusing for me: Can you add a comment to say that the case where `wafCtx == nil` means that the handle was destroyed?

Reply:
Added the following:
```
// Early return in the following cases:
// - wafCtx is nil, meaning it was concurrently released
// - err is not nil, meaning context creation failed
```